src/Database/Esqueleto/PostgreSQL/JSON.hs

```diff
@@ -0,0 +1,618 @@
+-- [true, "b", 5]         - -1          == [true, "b"]
+-- [true, "b", 5]         - -4          == [true, "b", 5]
+-- []                     - 1           == []
+-- {"1": true}            - 1           == ERROR: cannot delete from object using integer index
+-- 1                      - <anything>  == ERROR: cannot delete from scalar
+-- "a"                    - <anything>  == ERROR: cannot delete from scalar
+-- true                   - <anything>  == ERROR: cannot delete from scalar
+-- null                   - <anything>  == ERROR: cannot delete from scalar
+-- @
+--
+-- === __PostgreSQL Documentation__
+--
+-- /Works with any PostgreSQL of version >= 9.5/
+--
+-- @
+--    | Type    | Description                                                            |  Example
+-- ---+---------+------------------------------------------------------------------------+-------------------------------------------------
+--  - | text    | Delete key/value pair or string element from left operand.             | '{"a": "b"}'::jsonb - 'a'
+--    |         | Key/value pairs are matched based on their key value.                  |
+--  - | integer | Delete the array element with specified index (Negative integers count | '["a", "b"]'::jsonb - 1
+--    |         | from the end). Throws an error if top level container is not an array. |
+-- @
+--
+-- @since 3.1.0
+(-.) :: SqlExpr (Value (Maybe Aeson.Value))
+     -> Either Int Text
+     -> SqlExpr (Value (Maybe Aeson.Value))
+(-.) value (Right t) = unsafeSqlBinOp " - " value $ val t
+(-.) value (Left i) = unsafeSqlBinOp " - " value $ val i
+
+-- | Removes a set of keys from an object, or string elements from an array.
+-- This is the same operator internally as `-.`, but the option to use a @text
+-- array@, instead of @text@ or @integer@ was only added in version 10.
+-- That's why this function is seperate from `-.`.
+--
+-- NOTE: The following is equivalent:
+--
+-- @{some JSON expression} -. Right "a" -. Right "b"@
+--
+-- is equivalent to
+--
+-- @{some JSON expression} --. ["a","b"]@
+--
+-- === __PostgreSQL Documentation__
+--
+-- /Works with any PostgreSQL of version >= 10/
+--
+--    | Type    | Description                                                            |  Example
+-- ---+---------+------------------------------------------------------------------------+-------------------------------------------------
+--  - | text[]  | Delete multiple key/value pairs or string elements from left operand.  | '{"a": "b", "c": "d"}'::jsonb - '{a,c}'::text[]
+--    |         | Key/value pairs are matched based on their key value.                  |
+(--.) :: SqlExpr (Value (Maybe Aeson.Value))
+      -> [Text]
+      -> SqlExpr (Value (Maybe Aeson.Value))
+(--.) value = unsafeSqlBinOp " - " value . mkTextArray
+
+-- | This operator can remove elements nested in an object.
+-- If a 'Text' is not parsable as a number when selecting in an array
+-- (even when halfway through the selection).
```

Comment:
This sentence is unclear to me

Reply:
The `#-` operator removes a value nested (deep) within a JSON object/array. The RHO is a "path" to that value:

`v ^. JsonValue #-. ["user","name","last"]` would remove the "last" field in the object in the "name" field, which resides in the object that's in the "user" field of the top level object.

These examples show best what happens:

```
{"a": 3.14, "b": null}        #- ["a","b"] == {"a": 3.14, "b": null}
{"a": {"b":false}, "b": null} #- ["a","b"] == {"a": {}, "b": null}
```

The second sentence is unfinished, and should be:
"If a 'Text' is not parsable as a number when selecting in an array (even when halfway through the selection) an exception will be thrown"

Reply:
Does this explain it? And what do you think should I add to the description to make it more obvious? *without this elaborate explanation with examples*

Reply:
Finishing the second sentence clears it right up for me, thanks! :smile: